### ISSUE TITLE
The Dirichlet Process mixture model in the tutorial seems to just be a finite mixture mode

### ISSUE THREAD

**Issue by riversdark** [pyro-ppl/pyro#2613]
I'm looking at [the Dirichlet Process mixture model tutorial](http://pyro.ai/examples/dirichlet_process_mixture.html) but it seems to me that there is no real difference between this Dirichlet process formulation and the finite mixture model presented in the [Gaussian mixture model tutorial](http://pyro.ai/examples/gmm.html), bar the emission distributions of course.

In the Dirichlet Process mixture model we are supposed to have infinite number of clusters, but in the tutorial we have a fixed number of `T` clusters; I saw mentioned in the tutorial that this truncation does not affect our model because it is a simplification made in the inference stage to allow tractability, but how is this the case, given that the number of clusters `T` is already hard coded in the model definition?

The model uses a sticking breaking construction to generate the (supposed to be) infinite length cluster weights, but when truncated it's just the normal Dirichlet distribution, the number of inferred clusters will always be the same number `T`, and when the inference result looks sparse in the visualisation, it's just because we adjusted the hyperparameter alpha to favor fewer larger clusters, and removed all the smaller clusters from the visualisation. 

So basically I don't see anything in this model that's not already there in the finite mixture model. Am I missing something?

**Comment by fehiepsi** [comment]
@riversdark You are right that they are both finite mixture models. We don't have a tutorial for infinite version yet (it is an open question? or challenging? I guess). You can tract progress for the infinite version [here](https://github.com/pyro-ppl/pyro/issues/880).

**Comment by riversdark** [comment]
Ah I see. Thanks @fehiepsi !

**Comment by eb8680** [comment]
Closing in favor of #880. Note that truncation is a common and sensible strategy for inference in DPMMs, since we expect the number of components to grow much more slowly (roughly logarithmically) than the number of examples.
